extensions/purdue-course-search/src/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { useCallback, useState } from "react";
+
+import { ActionPanel, Action, List } from "@raycast/api";
+import { useStreamJSON } from "@raycast/utils";
+
+type SearchResultScheduleType = "Lecture" | "Distance Learning" | "Experiential" | "Laboratory" | "Individual Study";
+
+type SearchResult = {
+  id: string;
+  title: string;
+  subjectCode: string;
+  courseCode: string;
+  instructor: string[];
+  description: string;
+  capacity: number;
+  credits: [number] | [number, number];
+  term: string;
+  crn: number[];
+  sched: SearchResultScheduleType[];
+};
+
+export default function Command() {
+  const [searchText, setSearchText] = useState("");
+
+  const searchResultsFilter = useCallback(
```

Comment:
You can simplify by using [List](https://developers.raycast.com/api-reference/user-interface/list#list)'s `filtering` feature and specify [List.Item](https://developers.raycast.com/api-reference/user-interface/list#list.item)'s `keywords`. Or does raycast filtering not support your use-case?